packages/gatsby-source-drupal/src/gatsby-node.js

```diff
@@ -30,6 +33,55 @@ exports.sourceNodes = async (
     disallowedLinkTypes,
   } = pluginOptions
   const { createNode } = actions
+
+  if (webhookBody && Object.keys(webhookBody).length) {
+    const changesActivity = reporter.activityTimer(
+      `loading Drupal content changes`,
+      {
+        parentSpan,
+      }
+    )
+    changesActivity.start()
+
+    const { secret, action, id, data } = webhookBody
+    if (pluginOptions.secret && pluginOptions.secret !== secret) {
+      reporter.warn(
+        `The secret in this request did not match your plugin options secret.`
+      )
+      return
+    }
+    if (action === `delete`) {
+      actions.deleteNode({ node: getNode(createNodeId(id)) })
+      reporter.log(`Deleted node: ${id}`)
```

Comment:
Do we want to make this so explicit?
Maybe do something like contentful does where you give an overall update on what happened

https://github.com/gatsbyjs/gatsby/blob/00e08792ce08be904ff4a7a53c70ce70b7649f0b/packages/gatsby-source-contentful/src/gatsby-node.js#L155-L158

Reply:
It doesn't need to be so explicit. I just copied this code from the original `___updatePreview` endpoint, but it could be refactored to be a more general message.

Reply:
no need, you can do that another time